apis/io-engine/protobuf/v1/pool.proto

```diff
@@ -33,14 +34,34 @@ message PoolTypeValue {
 
 // Create pool arguments.
 message CreatePoolRequest {
-  string name = 1;           // name of the pool
-  google.protobuf.StringValue uuid = 2; // optional uuid for the pool to be created
-  repeated string disks = 3; // disk device paths or URIs to be claimed by the pool
-  PoolType pooltype = 4;     // type of the pool
+  // Name of the pool.
+  string name = 1;
+  // Optional uuid for the pool to be created.
+  google.protobuf.StringValue uuid = 2;
+  // Disk device paths or URIs to be claimed by the pool.
+  repeated string disks = 3;
+  // Type of the pool.
+  PoolType pooltype = 4;
   // Default cluster size is 4MiB (4 * 1024 * 1024), input cluster size must be in "bytes" but
   // must be multiple of 1MiB, else default cluster size i.e 4MiB will be considered. Minimum
   // cluster size can be input as 1048576 (1024 * 1024) bytes (1MiB)
   optional uint32 cluster_size = 5;
+  // Hint for how much of the pool capacity should be reserved for metadata.
+  // This hint is ratio of the desired to the pool backend's default reservation.
+  //
+  // 1 is the default value, and it means to use the pool backend's default
+  // reservation amount.
+  //
+  // 0.5 means an attempt to reserve two times less capacity for the metadata than
+  // the backend would do by default, trading replica/snapshot count
+  // for the user data space.
+  //
+  // 2 means a battempt to reserve two times more space for the metadata than
+  // the backend would do by default, allowing pool to accomodate more
+  // replicas/snapshots by reducing user data capacity.
+  //
+  // A pool backend is allowed to ignore this hint.
+  optional float md_resv_hint = 6;
```

Comment:
Is it possible that we set this value too high?
If so will return error code let us know this?

Reply:
My idea is this is "hint" kind of parameter: a backend is free to clamp to a usable value, or ignore it.
We can change this spec to that a backend _may_, or _must_ fail with EINVAL if the value is not usable. I'm not sure which is better here, _may_ or _must_.

Reply:
Ah you mean, if it's too large it just clamps it to the max it can for example? Seems acceptable I think.
What do you think as well @dsharma-dc

Reply:
I moved this MD arg into a separate optional message, and renamed it md_resv_ratio.

Reply:
Clamping sounds reasonable, but the question would be how high is `too high`. Instead of providing this hint in terms of metadata, I would suggest let's provide it in terms of `how much we expect pool to grow`. For example, If API field is set to 2, meaning we expect a 2x growth of lvstore in future, so we directly map this 2 as 200 into `num_md_pages_per_cluster_ratio` for spdk.